src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,125 @@
-import { Component } from 'react';
+import React, { useState, FormEvent } from 'react';
+import classNames from 'classnames';
 
-type Props = {};
-type State = {};
+interface Props {
+  onAdd: (movie: Movie) => void,
+}
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [invalidImgUrl, setInvalidImgUrl] = useState(false);
+  const [invalidImdbUrl, setInvalidImdbUrl] = useState(false);
 
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
-}
+  const inputCheck = (/^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/);
+
+  const checkImgValidity = () => {
+    if (!inputCheck.test(imgUrl)) {
+      setInvalidImgUrl(true);
+    }
+  };
+
+  const checkImdbValidity = () => {
+    if (!inputCheck.test(imdbUrl)) {
+      setInvalidImdbUrl(true);
+    }
+  };
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const newFilm = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    onAdd(newFilm);
+
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  return (
+    <form
+      className="Form"
+      onSubmit={(e) => {
+        handleSubmit(e);
+      }}
```

Comment:
just callback, if there are no arguments other than event

```suggestion
      onSubmit={handleSubmit}
```